tools/cli/src/mentat_cli/repl.rs

```diff
@@ -200,51 +232,99 @@ impl Repl {
                     Err(e) => eprintln!("{}", e.to_string()),
                 };
             },
-            Command::Close => self.close(),
-            Command::Query(query) => self.execute_query(query),
-            Command::QueryExplain(query) => self.explain_query(query),
+            Command::Query(query) => {
+                self.store
+                    .q_once(query.as_str(), None)
+                    .map_err(|e| e.into())
+                    .and_then(|o| {
+                        end = Some(PreciseTime::now());
+                        self.print_results(o)
+                    })
+                    .map_err(|err| {
+                        eprintln!("{:?}.", err);
+                    })
+                    .ok();
+            },
+            Command::QueryExplain(query) => {
+                self.explain_query(query);
+            },
+            Command::QueryPrepared(query) => {
+                self.store
+                    .q_prepare(query.as_str(), None)
+                    .and_then(|mut p| {
+                        let prepare_end = PreciseTime::now();
+                        if should_time {
+                            eprint_out("Prepare time");
+                            eprint!(": ");
+                            format_time(start.to(prepare_end));
+                        }
+                        // This is a hack.
+                        start = PreciseTime::now();
+                        let r = p.run(None);
+                        end = Some(PreciseTime::now());
+                        return r;
+                    })
+                    .map(|o| self.print_results(o))
+                    .map_err(|err| {
+                        eprintln!("{:?}.", err);
+                    })
+                    .ok();
+            },
             Command::Schema => {
                 let edn = self.store.conn().current_schema().to_edn_value();
                 match edn.to_pretty(120) {
                     Ok(s) => println!("{}", s),
                     Err(e) => eprintln!("{}", e)
                 };
-
-            }
-            Command::Transact(transaction) => self.execute_transact(transaction),
-            Command::Exit => {
-                self.close();
-                eprintln!("Exiting...");
-                process::exit(0);
+            },
+            Command::Sync(args) => {
+                match self.store.sync(&args[0], &args[1]) {
+                    Ok(_) => println!("Synced!"),
+                    Err(e) => eprintln!("{:?}", e)
+                };
             }
+            Command::Timer(on) => {
+                self.toggle_timer(on);
+            },
+            Command::Transact(transaction) => {
+                self.execute_transact(transaction);
+            },
         }
 
-        let end = PreciseTime::now();
+        let end = end.unwrap_or_else(PreciseTime::now);
         if should_time {
```

Comment:
nit: `should_time` is probably better named as `should_print_timings`? Timing happens regardless of this bool.